src/Analyzers/CSharp/CodeFixes/RemoveUnnecessaryNullableDirective/CSharpRemoveUnnecessaryNullableDirectiveCodeFixProvider.cs

```diff
@@ -53,10 +57,46 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             CodeActionOptionsProvider fallbackOptions,
             CancellationToken cancellationToken)
         {
-            foreach (var diagnostic in diagnostics)
+            var nullableDirectivesByNodes = diagnostics.GroupBy(x =>
+                x.Location.FindNode(findInsideTrivia: false, getInnermostNodeForTie: false, cancellationToken), x =>
+                x.Location.FindNode(findInsideTrivia: true, getInnermostNodeForTie: true, cancellationToken))
+                .ToDictionary(x => x.Key, x => x.ToList());
```

Comment:
what is thsi code for.  this needs docs.

Reply:
Added comments.